src/main/java/net/unicon/lti/security/lti/LTI3OAuthProviderProcessingFilter.java

```diff
@@ -68,76 +71,87 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
-                        httpServletRequest.setAttribute("LTI3", true); // indicate this request is an LTI3 one
-                        httpServletRequest.setAttribute("lti3_valid", lti3Request.isLoaded() && lti3Request.isComplete()); // is LTI3 request totally valid and complete
-                        httpServletRequest.setAttribute("lti3_message_type", lti3Request.getLtiMessageType()); // is LTI3 request totally valid and complete
-                        httpServletRequest.setAttribute(LTI3Request.class.getName(), lti3Request); // make the LTI3 data accessible later in the request if needed
+                log.debug("-------------------------------------------------------------------------------------------------------");
+                log.debug("Request Session Id in OAuthFilter: {}", httpServletRequest.getSession().getId());
+                log.debug("Request URL in OAuthFilter: {}", httpServletRequest.getRequestURL().toString());
+                log.debug("Request URI in OAuthFilter: {}", httpServletRequest.getRequestURI());
+                log.debug("Request Method in OAuthFilter: {}", httpServletRequest.getMethod());
+                log.debug("Request Cookies in OAuthFilter: {}", httpServletRequest.getCookies() != null ? Arrays.stream(httpServletRequest.getCookies()).toList().toString() : null);
+                Cookie[] cookies = httpServletRequest.getCookies();
+                if (cookies != null) {
+                    for (Cookie cookie : cookies) {
+                        log.debug("Cookie name: {}", cookie.getName());
+                        log.debug("Cookie value: {}", cookie.getValue());
                     }
                 }
+            }
 
-                filterChain.doFilter(servletRequest, servletResponse);
-
-                this.resetAuthenticationAfterRequest();
-            } catch (ExpiredJwtException eje) {
-                log.info("Security exception for user {} - {}", eje.getClaims().getSubject(), eje.getMessage());
-                ((HttpServletResponse) servletResponse).setStatus(HttpServletResponse.SC_UNAUTHORIZED);
-                log.debug("Exception " + eje.getMessage(), eje);
-            } catch (SignatureException ex) {
-                log.info("Invalid JWT signature: {0}", ex.getMessage());
-                log.debug("Exception " + ex.getMessage(), ex);
-                ((HttpServletResponse) servletResponse).setStatus(HttpServletResponse.SC_UNAUTHORIZED);
-            } catch (DataServiceException e) {
-                log.error("Error in the Data Service", e);
+            // First we validate that the state is a good state.
+
+            //First, we make sure that the query has a state
+            String state = httpServletRequest.getParameter("state");
+            String link = httpServletRequest.getParameter("link");
+            if (httpServletRequest.getSession().getAttribute("lti_state") == null) {
```

Comment:
Is this okay or was this the thing they're moving, long term, into local storage hosted by the LMS?

Reply:
The state is what will ultimately get saved in an iframe hosted by the LMS as part of the cookie solution down the road. Canvas (and other LMSs) haven't yet implemented the LMS storage solution, but it is coming soon.

Reply:
Yep - thank you for the full explanation there!